source/m2mconnectionhandlerpimpl.cpp

```diff
@@ -434,6 +435,7 @@ bool M2MConnectionHandlerPimpl::resolve_hostname(const char *address,
     if(_socket_server > 0) {
         close(_socket_server);
         _socket_server = -1;
+        pthread_join(_listen_thread, NULL);
```

Comment:
I am not completely sure, if _listen_thread needs to be cleared here too to avoid the pthread_cancel()+pthread_join() sequence at destructor.